### ISSUE TITLE
Class 'Swap\Swap' not found in SwapServiceProvider.php:168

### ISSUE THREAD

**Issue by simplenotezy** [florianv/laravel-swap#10]
For some strange reason, I get the following error. I think it happened after updating from composer:

```
Class 'Swap\Swap' not found in SwapServiceProvider.php:168
```

Stacktrace:

```
[2016-03-28 14:14:21] local.ERROR: Symfony\Component\Debug\Exception\FatalThrowableError: Fatal error: Class 'Swap\Swap' not found in /home/vagrant/Projects/myapp/vendor/florianv/laravel-swap/src/SwapServiceProvider.php:168
Stack trace:
#0 /home/vagrant/Projects/myapp/vendor/laravel/framework/src/Illuminate/Container/Container.php(735): Florianv\LaravelSwap\SwapServiceProvider->Florianv\LaravelSwap\{closure}(Object(Illuminate\Foundation\Application), Array)
#1 /home/vagrant/Projects/myapp/vendor/laravel/framework/src/Illuminate/Container/Container.php(633): Illuminate\Container\Container->build(Object(Closure), Array)
#2 /home/vagrant/Projects/myapp/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(687): Illuminate\Container\Container->make('swap', Array)
#3 /home/vagrant/Projects/myapp/vendor/laravel/framework/src/Illuminate/Container/Container.php(1178): Illuminate\Foundation\Application->make('swap')
#4 /home/vagrant/Projects/myapp/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(151): Illuminate\Container\Container->offsetGet('swap')
#5 /home/vagrant/Projects/myapp/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(120): Illuminate\Support\Facades\Facade::resolveFacadeInstance('swap')
#6 /home/vagrant/Projects/myapp/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(207): Illuminate\Support\Facades\Facade::getFacadeRoot()
#7 /home/vagrant/Projects/myapp/app/Currency.php(73): Illuminate\Support\Facades\Facade::__callStatic('quote', Array)
```

**Comment by simplenotezy** [comment]
I went to a older branch and ran composer install. I noticed the following from the composer status:

```
Removing florianv/swap (v2.4.0)
  - Installing florianv/swap (v2.3.1)
    Downloading: 100%
```

Now the issue has vanished. I am pulling in:

```
"florianv/laravel-swap": "0.*",
```

**Comment by florianv** [comment]
I don't really know what could have happened as we didn't modify the `SwapServiceProvider` since the first release of `Swap`..so it must be related to `Laravel`. 

Which Laravel version are you using ?

**Comment by simplenotezy** [comment]
It works on laravel 5.2.23, but breaks on 5.2.26.

**Comment by simplenotezy** [comment]
However, it seems to be working just fine now. After running composer update.

Will report if I see this issue again.

**Comment by florianv** [comment]
Ok thanks for the update. It was probably a cache issue

**Comment by farzanbd** [comment]
I am having the same issue on the newly released Laravel 5.8

**Comment by farzanbadakhshan** [comment]
I'm having the same issue on Laravel 5.8.
